sentry_sdk/integrations/django/__init__.py

```diff
@@ -370,6 +373,15 @@ def _set_transaction_name_and_source(scope, transaction_style, request):
             transaction_name,
             source=source,
         )
+    except Resolver404:
+        urlconf = import_module(settings.ROOT_URLCONF)
+        # This exception only gets thrown when transaction_style is `function_name`
+        # So we don't check here what style is configured
+        if hasattr(urlconf, "handler404"):
+            handler = urlconf.handler404
+            scope.transaction = transaction_from_function(
+                getattr(handler, "view_class", handler)
+            )
```

Comment:
```suggestion
            scope.transaction = urlconf.handler404
```

I guess it would suffice to just set the handler404 as the transaction name. As the handler404 is already the dotted path to the function. (`transaction_from_function` tries to get a dotted path from a given function object. In my demo project it returned none, because `getattr(handler, "view_class", handler)` just returned the handler404 string)

Reply:
`urlconf.handler404` doesn't have to be a dotted path to a function, especially when using a class based view it can also be an actual function call of `ViewClass.as_view()`

Reply:
ok, got it. I have changed now the code to take the handler as-is if it is a string type and otherwise try to get the `transaction_from_function`